src/app/lab/components/units-dropdown.component.ts

```diff
@@ -0,0 +1,44 @@
+import { Component, Input, Output, EventEmitter } from '@angular/core';
+
+import { Chart } from '../../models/chart.model';
+import { Indicator } from '../../models/indicator.model';
+
+/*  Units Dropdown Component
+
+    -- Requires handling unit selection
+    Expected use:
+        <ccl-units-dropdown
+            [indicator]="your_indicator"
+            [unit]="your_unit"
+            (unitSelected)="onUnitSelected($event)">
+*/
+
+@Component({
+  selector: 'ccl-units-dropdown',
+  template: `<div dropdown class="dropdown dropdown-units">
+              <button dropdownToggle class="button dropdown-toggle" type="button"
+                id="unitsDropdown" data-toggle="dropdown" aria-haspopup="true"
+                aria-expanded="true">
+                <i class="icon-flash"></i>
+                {{ unit || indicator.default_units }}
+                <i class="icon-angle-down caret"></i>
+              </button>
+              <ul *dropdownMenu class="dropdown-menu" aria-labelledby="unitsDropdown">
+                <li *ngFor="let unit of indicator.available_units">
+                  <a (click)="onUnitSelected(unit)"
+                    placement="bottom">{{ unit }}</a>
+                </li>
+              </ul>
+            </div>`
+})
+
+export class UnitsDropdownComponent {
+
+    @Input() indicator: Indicator;
```

Comment:
I had thought I'd needed `indicator.default_unit` in addition to `indicator.available_units,` so i sent through indicator. Turns out no, just `indicator.available_units` because I set `chart.unit` upon chart initialization.